source/client.c

```diff
@@ -1527,6 +1527,12 @@ uint16_t aws_mqtt_resubscribe_existing_topics(
     static const size_t sub_size = sizeof(struct aws_mqtt_topic_subscription);
 
     if (sub_count == 0) {
+        aws_raise_error(AWS_ERROR_MQTT_NO_TOPICS_FOR_RESUBSCRIBE);
+        AWS_LOGF_WARN(
+            AWS_LS_MQTT_CLIENT,
+            "id=%p: Resubscribe is unnecessary, error %s.",
```

Comment:
This reads weirdly. How about "There are currently no open subscriptions, resubscribe is unnecessary and no-op."